xbmc/cores/dvdplayer/DVDPlayer.cpp

```diff
@@ -1108,6 +1108,34 @@ void CDVDPlayer::CheckBetterStream(CCurrentStream& current, CDemuxStream* stream
     OpenStream(current, stream->iId, stream->source);
 }
 
+int CDVDPlayer::PreloadFileInfo(const CFileItem& file, const CPlayerOptions &options)
+{
+  m_bAbortRequest = false;
+  SetPlaySpeed(DVD_PLAYSPEED_NORMAL);
+
+  m_State.Clear();
```

Comment:
I am not sure I understand what you mean by "modified from outside"? PreloadFileInfo() is mimicking the OpenFile() method to open a media file. It is never used on a actually playing media. The player used to access the file info is not referenced outside of GUIDialogVideoInfo.cpp, it is NOT the application player.